<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 3: Microbiome statistics and visualizations | Beginner Microbiome Analysis 2025</title>
  <meta name="description" content="CBW’s Bookdown Template for Workshops" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 3: Microbiome statistics and visualizations | Beginner Microbiome Analysis 2025" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://cbw-dev.github.io/bookdown-template/img/bioinformatics_logo.png" />
  <meta property="og:description" content="CBW’s Bookdown Template for Workshops" />
  <meta name="github-repo" content="cbw-dev/BMB_2025" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 3: Microbiome statistics and visualizations | Beginner Microbiome Analysis 2025" />
  
  <meta name="twitter:description" content="CBW’s Bookdown Template for Workshops" />
  <meta name="twitter:image" content="https://cbw-dev.github.io/bookdown-template/img/bioinformatics_logo.png" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-1.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>

 <li><a href="./">Beginner Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="course-schedule.html"><a href="course-schedule.html"><i class="fa fa-check"></i>Course Schedule</a>
<ul>
<li class="chapter" data-level="" data-path="course-schedule.html"><a href="course-schedule.html#meet-your-faculty"><i class="fa fa-check"></i>Meet Your Faculty</a></li>
</ul></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1.html"><a href="module-1.html"><i class="fa fa-check"></i>Module 1</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html"><i class="fa fa-check"></i>Module 3: Microbiome statistics and visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#table-of-contents"><i class="fa fa-check"></i>Table of Contents</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#build-tree-with-sepp-qiime-2-plugin"><i class="fa fa-check"></i>1. Build tree with SEPP QIIME 2 plugin</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#rarefaction-curves"><i class="fa fa-check"></i>2. Generate rarefaction curves</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#diversity-ordination"><i class="fa fa-check"></i>3. Calculating diversity metrics and generating ordination plots</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#stacked-bar-chart"><i class="fa fa-check"></i>4. Generate stacked bar chart of taxa relative abundances</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#ancom"><i class="fa fa-check"></i>5. Identifying differentially abundant features with ANCOM</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#core-microbiome"><i class="fa fa-check"></i>6. Identifying shared features across groups of samples</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#heatmaps"><i class="fa fa-check"></i>7. Heatmaps</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#exporting-data"><i class="fa fa-check"></i>8. Exporting data from QIIME 2 for use in other software</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#id_18s-its"><i class="fa fa-check"></i>9. Considerations for 18S and ITS data</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#building-a-de-novo-tree-for-18s"><i class="fa fa-check"></i>Building a de novo tree for 18S</a></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#building-the-its-tree"><i class="fa fa-check"></i>Building the ITS tree</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-microbiome-statistics-and-visualizations.html"><a href="module-3-microbiome-statistics-and-visualizations.html#answers"><i class="fa fa-check"></i>Answers</a></li>
</ul></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
 <center>
 <a href="https://www.cihr-irsc.gc.ca"><img src="img/sponsors/cihr.png" width=90% style="padding-bottom: 20px" alt="Canadian Institutes of Health Research logo."></a>

 <a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.webp" width=40% style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
 <a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" width=40% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Genomics logo.">
 <br><br>
 <a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=25% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">
 <a href="https://www.hpcforhealth.ca/"><img src="img/sponsors/hpc4health.png" width=60% style="display:inline-block;" style="padding-bottom: 20px "alt="HPC4Health logo.">
 <br><br>

 <li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Beginner Microbiome Analysis 2025</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-3-microbiome-statistics-and-visualizations" class="section level1 hasAnchor">
<h1>Module 3: Microbiome statistics and visualizations<a href="module-3-microbiome-statistics-and-visualizations.html#module-3-microbiome-statistics-and-visualizations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Material by Juan Santana</p>
<p>This tutorial is part of the 2025 Canadian Bioinformatics Workshops Beginner Microbiome Analysis (Vancouver, BC, May 26-27). It is based on the Amplicon SOP v2 available on the <a href="https://github.com/LangilleLab/microbiome_helper/wiki/Amplicon-SOP-v2-(qiime2-2022.11)">Microbiome Helper repository</a> and previous workshops designed by Diana Haider, Robert Beiko and Monica Alvaro Fuss.</p>
<div id="table-of-contents" class="section level2 hasAnchor">
<h2>Table of Contents<a href="module-3-microbiome-statistics-and-visualizations.html#table-of-contents" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><a href="module-3-microbiome-statistics-and-visualizations.html#build-tree">Build tree</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#rarefaction-curves">Generate rarefaction curves</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#diversity-ordination">Calculating diversity metrics and generating ordination plots</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#stacked-bar-chart">Generate stacked bar chart of taxa relative abundances</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#ancom">Identifying differentially abundant features with ANCOM</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#core-microbiome">Core microbiome analysis</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#heatmaps">Heatmaps</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#exporting-data">Exporting data from QIIME 2 for use in other software</a><br />
</li>
<li><a href="module-3-microbiome-statistics-and-visualizations.html#id_18s-its">Considerations for 18S and ITS data</a></li>
</ol>
</div>
<div id="introduction" class="section level2 hasAnchor">
<h2>Introduction<a href="module-3-microbiome-statistics-and-visualizations.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Modules 2 and 3 offer a step-by-step walkthrough of an end-to-end pipeline for analyzing high-throughput marker gene data using the command-line interface. Commonly used marker genes in microbiome research include 16S ribosomal RNA (rRNA) for prokaryotes, 18S rRNA for eukaryotes, and the internal transcribed spacer (ITS) region for fungi.</p>
<p>In this tutorial, we primarily focus on the same 16S rRNA dataset derived from wild blueberry (<em>Vaccinium angustifolium</em>) soil communities from natural and managed habitats we processed in Module 2. At the end of the tutorial, we also provide guidance on processing the 18S rRNA dataset from plastics incubated in a coastal marine environment (plastisphere), as well as the ITS dataset from the gut mycobiome (i.e., fungi) during pregnancy.</p>
<p>You can learn more about these studies in the following publications:</p>
<ul>
<li><a href="https://apsjournals.apsnet.org/doi/10.1094/PBIOMES-03-17-0012-R">Variation in Bacterial and Eukaryotic Communities Associated with Natural and Managed Wild Blueberry Habitats</a></li>
<li><a href="https://www.frontiersin.org/articles/10.3389/fmicb.2019.01682/full#B50">Metagenomic Functional Shifts to Plant Induced Environmental Changes</a>
-<a href="https://www.sciencedirect.com/science/article/pii/S0025326X22003836#s0050">Microbial pioneers of plastic colonisation in coastal seawaters</a></li>
<li><a href="https://gut.bmj.com/content/73/8/1302.long">Landscape of the gut mycobiome dynamics during pregnancy and its relationship with host metabolism and pregnancy health</a></li>
</ul>
<p>In Module 2, we covered the fundamentals of marker gene analysis, from processing raw reads to generating a filtered feature table. In Module 3, we will explore examples of downstream analyses used to draw biological insights from these data. The workflow described is integrated into the latest release of QIIME 2 (Quantitative Insights into Microbial Ecology, version 2025.4). As introduced in Module 2, this widely used microbiome bioinformatics platform is built around user-developed software packages called plugins, which operate on QIIME 2 artifact files (with the .qza extension). Documentation for these plugins—including tutorials and additional resources—can be found in the <a href="https://amplicon-docs.qiime2.org/en/latest/">QIIME 2 user documentation</a>. QIIME 2 also offers interpretable visualizations, which can be viewed by opening .qzv files in <a href="https://view.qiime2.org/">QIIME2 View</a></p>
<p>Let’s set up our Module 3 directory inside <code>workspace</code> and create a symlink to the data we will be using (from Module 2).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb1-1" tabindex="-1"></a><span class="bu">cd</span> ~/workspace</span>
<span id="cb1-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb1-2" tabindex="-1"></a><span class="fu">mkdir</span> bmb_module3 bmb_module3/16S_analysis</span>
<span id="cb1-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb1-3" tabindex="-1"></a><span class="bu">cd</span> bmb_module3/16S_analysis</span>
<span id="cb1-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb1-4" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/16S_analysis/deblur_output .</span>
<span id="cb1-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb1-5" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/16S_analysis/taxa .</span>
<span id="cb1-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb1-6" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/16S_analysis/Blueberry_16S_metadata.tsv .</span></code></pre></div>
<p>If you deactivated your QIIME2 environment, reactivate it with the command below.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb2-1" tabindex="-1"></a><span class="ex">conda</span> activate qiime2-amplicon-2025.4</span></code></pre></div>
<p>When you are finished this tutorial you can deactivate the conda environment using:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb3-1" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div>
<p>Throughout this module, there are some questions aimed to help your understanding of some of the key concepts. You’ll find the answers at the bottom of this page, but no one will be marking them.</p>
</div>
<div id="build-tree-with-sepp-qiime-2-plugin" class="section level2 hasAnchor">
<h2>1. Build tree with <a href="https://amplicon-docs.qiime2.org/en/latest/references/plugins/fragment-insertion.html#q2-plugin-fragment-insertion" id="build-tree">SEPP QIIME 2 plugin</a><a href="module-3-microbiome-statistics-and-visualizations.html#build-tree-with-sepp-qiime-2-plugin" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5904434/">SEPP</a> (SATé-enabled Phylogenetic Placement) is a tool used to place short DNA sequences—such as 16S rRNA amplicon sequence variants (ASVs)—into an existing, high-quality reference phylogenetic tree. This is particularly helpful when you are working with microbiome data and want to infer evolutionary relationships more accurately. We will use QIIME 2’s <code>q2-fragment-insertion</code> plugin to place ASVs derived from our <strong>16S data</strong> into a reference phylogenetic tree using the command below.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb4-1" tabindex="-1"></a><span class="ex">qiime</span> fragment-insertion sepp</span>
<span id="cb4-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb4-2" tabindex="-1"></a>  <span class="ex">--i-representative-sequences</span> deblur_output/rep_seqs_final.qza <span class="dt">\</span></span>
<span id="cb4-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb4-3" tabindex="-1"></a>  <span class="at">--i-reference-database</span> ~/CourseData/MIC_data/bmb_module3/16S_analysis/sepp-refs-gg-13-8.qza <span class="dt">\</span></span>
<span id="cb4-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb4-4" tabindex="-1"></a>  <span class="at">--o-tree</span> asvs-tree.qza <span class="dt">\</span></span>
<span id="cb4-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb4-5" tabindex="-1"></a>  <span class="at">--o-placements</span> insertion-placements.qza <span class="dt">\</span></span>
<span id="cb4-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb4-6" tabindex="-1"></a>  <span class="at">--p-threads</span> 4</span></code></pre></div>
<p><strong>Due to memory constraints, you can instead copy the output into your folder with the following command:</strong></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb5-1" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module3/output/16S_analysis/asvs-tree.qza .</span>
<span id="cb5-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb5-2" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module3/output/16S_analysis/insertion-placements.qza .</span></code></pre></div>
<p>High-quality reference phylogenetic trees can be downloaded from <a href="https://docs.qiime2.org/2024.10/data-resources/">QIIME2’s data resources</a>. However, it is important to ensure that the reference tree used for sequence placement matches the same reference database used for taxonomic classification (e.g., Greengenes or SILVA). In our workflow, we use <code>sepp-refs-gg-13-8.qza</code>, as our sequences were classified using the Greengenes database. Alternatively, custom reference files can be specified for placing other types of amplicons. However, for marker genes such as <strong>18S and ITS</strong>, the recommended approach is to construct a <em>de novo</em> phylogenetic tree, as outlined in section 9. <a href="#id_9-considerations-for-18s-and-its-data">Considerations for 18S and ITS data</a> and and further detailed in the <a href="https://github.com/LangilleLab/microbiome_helper/wiki/Amplicon-SOP-v2-(qiime2-2022.11)">Microbiome Helper repository</a>.</p>
</div>
<div id="rarefaction-curves" class="section level2 hasAnchor">
<h2>2. Generate rarefaction curves<a href="module-3-microbiome-statistics-and-visualizations.html#rarefaction-curves" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After inferring ASVs from our sequence reads, we can generate <strong>rarefaction curves</strong> to evaluate whether our sequencing depth was sufficient to capture most of the microbial diversity in each sample. The command below will generate these plots. The <code>--p-max-depth</code> parameter sets the maximum sequencing depth to sample across. Inspect the <code>deblur_table_final_summary.qzv</code> file (created in Module 2) using <a href="https://view.qiime2.org/">QIIME2 View</a> to determine this number. For our 16S dataset, the sample with the highest number of reads contains 8,650 sequences, so we’ll use that as the max depth.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb6-1" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-rarefaction <span class="dt">\</span></span>
<span id="cb6-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb6-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb6-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb6-3" tabindex="-1"></a>  <span class="at">--p-max-depth</span> 8650 <span class="dt">\</span></span>
<span id="cb6-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb6-4" tabindex="-1"></a>  <span class="at">--p-steps</span> 20 <span class="dt">\</span></span>
<span id="cb6-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb6-5" tabindex="-1"></a>  <span class="at">--i-phylogeny</span> asvs-tree.qza <span class="dt">\</span></span>
<span id="cb6-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb6-6" tabindex="-1"></a>  <span class="at">--o-visualization</span> rarefaction_curves.qzv</span></code></pre></div>
<p>This will produce a <code>.qzv</code> file that can be opened in QIIME 2 View to explore the number of observed features (as well as two alpha diversity metrics: Shannon index and Faith’s phylogenetic diversity) across rarefied sequencing depths for each sample.</p>
<p><strong>Question 1: What is a good rarefaction depth for diversity analysis?</strong></p>
</div>
<div id="diversity-ordination" class="section level2 hasAnchor">
<h2>3. Calculating diversity metrics and generating ordination plots<a href="module-3-microbiome-statistics-and-visualizations.html#diversity-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>QIIME 2 can calculate commonly used alpha and beta diversity metrics—including Faith’s Phylogenetic Diversity, Shannon diversity, and UniFrac distances—using a single command. Ordination plots (e.g., PCoA plots based on weighted UniFrac distances) are also generated automatically. Before calculating these metrics, QIIME 2 rarefies all samples to the same sequencing depth to ensure fair comparisons. The <code>--p-sampling-depth</code> parameter defines this cutoff. Any samples with fewer reads than this threshold will be excluded from the analysis. Based on the rarefaction curves we generated earlier, we determined that the lowest reasonable sequencing depth across all samples is <strong>3,432</strong> reads. Using this value ensures we retain all samples while still capturing most of the microbial diversity.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb7-1" tabindex="-1"></a><span class="ex">qiime</span> diversity core-metrics-phylogenetic <span class="dt">\</span></span>
<span id="cb7-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb7-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb7-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb7-3" tabindex="-1"></a>  <span class="at">--i-phylogeny</span> asvs-tree.qza <span class="dt">\</span></span>
<span id="cb7-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb7-4" tabindex="-1"></a>  <span class="at">--p-sampling-depth</span> 3432  <span class="dt">\</span></span>
<span id="cb7-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb7-5" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb7-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb7-6" tabindex="-1"></a>  <span class="at">--p-n-jobs-or-threads</span> 4 <span class="dt">\</span></span>
<span id="cb7-7"><a href="module-3-microbiome-statistics-and-visualizations.html#cb7-7" tabindex="-1"></a>  <span class="at">--output-dir</span> diversity</span></code></pre></div>
<p>This command will output a <code>diversity/</code> folder containing:</p>
<p><em>Alpha diversity vectors (e.g., Shannon, Faith’s PD)
</em>Beta diversity distance matrices (e.g., UniFrac, Bray-Curtis)
<em>Rarefied feature table
</em>PCoA ordination plots (<code>emperor.qzv</code> files)</p>
<p>You can explore these ordination plots to visually assess whether microbial communities cluster based on sample groupings (e.g., natural vs. managed habitats).However, to statistically and visually compare alpha diversity across sample categories (e.g., habitat type), you have to generate boxplots and perform statistical tests using the following command:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb8-1" tabindex="-1"></a><span class="ex">qiime</span> diversity alpha-group-significance <span class="dt">\</span></span>
<span id="cb8-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb8-2" tabindex="-1"></a>  <span class="at">--i-alpha-diversity</span> diversity/shannon_vector.qza <span class="dt">\</span></span>
<span id="cb8-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb8-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb8-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb8-4" tabindex="-1"></a>  <span class="at">--o-visualization</span> diversity/shannon_compare_groups.qzv</span></code></pre></div>
<p><em>Hint:To analyze other alpha diversity metrics, simply replace <code>shannon_vector.qza</code> with another file from the <code>diversity/</code> directory. To view available metrics, run:</em></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb9-1" tabindex="-1"></a><span class="fu">ls</span> diversity/<span class="pp">*</span>_vector.qza</span></code></pre></div>
<p>Note that you can also export this or any other diversity metric file (ending in <em>.qza</em>) using <code>qiime tools export</code> and analyze them with a different program (e.g., R, Python, Excel).</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb10-1" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="dt">\</span></span>
<span id="cb10-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb10-2" tabindex="-1"></a>  <span class="at">--input-path</span> diversity/shannon_vector.qza <span class="dt">\</span></span>
<span id="cb10-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb10-3" tabindex="-1"></a>  <span class="at">--output-path</span> exported_shannon</span></code></pre></div>
<p>This will produce a file called <code>alpha-diversity.tsv</code> inside the <code>exported_shannon/</code> folder.</p>
<p><strong>Question 2: are there any significant differences in alpha diversity between any of our metadata categories?</strong></p>
<p><strong>Question 3: which metadata category appears to provide more separation in the beta diversity PCoA plots?</strong></p>
<p>If you want to run a PERMANOVA test to assess whether beta diversity differs significantly between groups, you can use the following command. Be sure to update the command with the specific beta diversity metric you’re analyzing (e.g., Bray-Curtis, UniFrac) and replace category with the appropriate column name from your metadata file that defines your groups of interest.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb11-1" tabindex="-1"></a><span class="ex">qiime</span> diversity beta-group-significance <span class="dt">\</span></span>
<span id="cb11-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb11-2" tabindex="-1"></a>  <span class="at">--i-distance-matrix</span> diversity/bray_curtis_distance_matrix.qza <span class="dt">\</span></span>
<span id="cb11-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb11-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb11-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb11-4" tabindex="-1"></a>  <span class="at">--m-metadata-column</span> Category <span class="dt">\</span></span>
<span id="cb11-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb11-5" tabindex="-1"></a>  <span class="at">--o-visualization</span> diversity/bray_curtis_compare_groups_category.qzv</span></code></pre></div>
<p>This will produce a <code>.qzv</code> file containing PERMANOVA results and boxplots for visualizing distances within and between groups.</p>
<p><strong>Question 4: what do you mean I’m getting an error message? It should be working fine, it worked last time! I can’t see what I’m doing wro… oooooh there we go, I see now :)</strong></p>
</div>
<div id="stacked-bar-chart" class="section level2 hasAnchor">
<h2>4. Generate stacked bar chart of taxa relative abundances<a href="module-3-microbiome-statistics-and-visualizations.html#stacked-bar-chart" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A useful way to visualize the composition of microbial communities across samples is through interactive stacked bar charts of taxonomic relative abundances. This type of plot allows you to explore differences in microbial composition at various taxonomic levels (e.g., phylum, genus) across experimental groups. You can generate the visualization using the command below.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb12-1" tabindex="-1"></a><span class="ex">qiime</span> taxa barplot <span class="dt">\</span></span>
<span id="cb12-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb12-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb12-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb12-3" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> taxa/classification.qza <span class="dt">\</span></span>
<span id="cb12-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb12-4" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb12-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb12-5" tabindex="-1"></a>  <span class="at">--o-visualization</span> taxa_barplot.qzv</span></code></pre></div>
<p>The interactive <code>.qzv</code> file generated with this command can be inspected with the QIIME2 viewer. In the plot you can explore dominant taxa across samples or groups, navigate between taxonomic levels (kingdom to species) and compare relative abundances between metadata-defined sample groups.</p>
<p><em>Hint: your metadata file includes any grouping columns you want to use to color or group the samples in the visualization (e.g., treatment, site, time point).</em></p>
<p><strong>Question 5: can you identify any patterns between the metadata groups?</strong></p>
</div>
<div id="ancom" class="section level2 hasAnchor">
<h2>5. Identifying differentially abundant features with ANCOM<a href="module-3-microbiome-statistics-and-visualizations.html#ancom" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4450248/">ANCOM</a> (Analysis of Composition of Microbiomes) is a statistical method used to identify taxa (or features) that differ significantly in relative abundance between sample groups. ANCOM is robust to the compositional nature of microbiome data and does not assume a particular distribution for the features. However, ANCOM requires that all features in the table be non-zero, so a pseudocount (commonly set to 1) must first be added to avoid issues with zeros in the count matrix. The command below creates a new feature table where a count of 1 is added to all zero entries.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb13-1" tabindex="-1"></a><span class="ex">qiime</span> composition add-pseudocount <span class="dt">\</span></span>
<span id="cb13-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb13-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb13-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb13-3" tabindex="-1"></a>  <span class="at">--p-pseudocount</span> 1 <span class="dt">\</span></span>
<span id="cb13-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb13-4" tabindex="-1"></a>  <span class="at">--o-composition-table</span> deblur_output/deblur_table_final_pseudocount.qza</span></code></pre></div>
<p>We then run ANCOM using the command below. <code>--m-metadata-column</code> defines the groups you’d like to compare (e.g., Treatment, Site, Timepoint, etc.) according to the columns in your metadata file.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb14-1" tabindex="-1"></a><span class="ex">qiime</span> composition ancom <span class="dt">\</span></span>
<span id="cb14-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb14-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final_pseudocount.qza <span class="dt">\</span></span>
<span id="cb14-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb14-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb14-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb14-4" tabindex="-1"></a>  <span class="at">--m-metadata-column</span> category <span class="dt">\</span></span>
<span id="cb14-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb14-5" tabindex="-1"></a>  <span class="at">--output-dir</span> ancom_output</span></code></pre></div>
<p>The results from ANCOM will be saved in the <code>ancom_output/</code> directory.The key file is ancom.qzv, which can be viewed interactively at QIIME 2 View. The output highlights features (e.g., ASVs, OTUs, or taxa) that are differentially abundant between the specified groups based on their W statistic (number of significant pairwise comparisons).</p>
<p><em>Hint: ANCOM is sensitive to sparsity in the data. Consider filtering low-abundance or low-frequency features before running ANCOM to improve detection power and reduce false positives. Alternatively, we can use ANCOM-BC (see below)</em></p>
<p>ANCOM-BC addresses limitations in the original ANCOM method by incorporating bias correction and estimating absolute abundances from compositional data. It performs better under sparsity and unequal library sizes. Note that the <code>--p-formula</code> parameter lets you specify covariates or design models (e.g., Treatment + Timepoint), in our workflow we only test for significant abundances based on the “category” variable. In addition, we supply the original raw counts instead of the compositionally transformed table.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb15-1" tabindex="-1"></a><span class="ex">qiime</span> composition ancombc <span class="dt">\</span></span>
<span id="cb15-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb15-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb15-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb15-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb15-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb15-4" tabindex="-1"></a>  <span class="at">--p-formula</span> category <span class="dt">\</span></span>
<span id="cb15-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb15-5" tabindex="-1"></a>  <span class="at">--output-dir</span> ancombc_output/ </span></code></pre></div>
<p>While <code>qiime composition ancombc</code> produces a <code>.qza</code> file of differentials (log-fold changes and statistics), QIIME 2 does not currently provide built-in commands to visualize this artifact as a <code>.qzv</code>. Therefore, we use another command to generate visualizations.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb16-1" tabindex="-1"></a><span class="ex">qiime</span> composition da-barplot <span class="dt">\</span></span>
<span id="cb16-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb16-2" tabindex="-1"></a>  <span class="at">--i-data</span> ancombc_output/differentials.qza <span class="dt">\</span></span>
<span id="cb16-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb16-3" tabindex="-1"></a>  <span class="at">--p-significance-threshold</span> 0.05 <span class="dt">\</span></span>
<span id="cb16-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb16-4" tabindex="-1"></a>  <span class="at">--p-effect-size-threshold</span> 1.0 <span class="dt">\</span></span>
<span id="cb16-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb16-5" tabindex="-1"></a>  <span class="at">--o-visualization</span> ancombc_output/ancombc_results.qzv</span></code></pre></div>
<p>The resultant <code>.qzv</code> barplot will show you features significantly enriched in one or another condition.</p>
<p><strong>Question 6: Does ANCOM identify any differentially abundant taxa between any of the metadata groups? If so, which one(s)? Is the output from ANCOM-BC any different?</strong></p>
</div>
<div id="core-microbiome" class="section level2 hasAnchor">
<h2>6. Identifying shared features across groups of samples<a href="module-3-microbiome-statistics-and-visualizations.html#core-microbiome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The core microbiome refers to the set of microbial taxa consistently found across a group of samples or within specific treatment groups. Identifying core taxa helps understand the stable and potentially important microbes in your system. In QIIME2 we can filter features by prevalence across samples using the command below. <code>-p-min-fraction 0.8</code> means features must be present in 80% of samples to be considered “core”.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb17-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table core-features <span class="dt">\</span></span>
<span id="cb17-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb17-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb17-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb17-3" tabindex="-1"></a>  <span class="at">--p-min-fraction</span> 0.8 <span class="dt">\</span></span>
<span id="cb17-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb17-4" tabindex="-1"></a>  <span class="at">--o-visualization</span> core_features.qzv</span></code></pre></div>
<p>This will output a feature table summary with the fraction of the ASVs shared across all samples. The core microbiome plugin <code>qiime feature-table core-features</code> doesn’t have a an option to group samples by treatment, so, if you want to identify the shared ASVs across samples from a group (e.g. managed) you need to subset your feature table first by treatment group using <code>qiime feature-table filter-samples</code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb18-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table filter-samples <span class="dt">\</span></span>
<span id="cb18-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb18-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb18-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb18-3" tabindex="-1"></a>  <span class="at">--m-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb18-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb18-4" tabindex="-1"></a>  <span class="at">--p-where</span> <span class="st">&quot;[category]=&#39;Managed&#39;&quot;</span> <span class="dt">\</span></span>
<span id="cb18-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb18-5" tabindex="-1"></a>  <span class="at">--o-filtered-table</span> table_managed.qza</span></code></pre></div>
<p>We can then run <code>qiime feature-table core-features</code> on the filtered table <code>table_managed.qza</code> to identify shared ASVs across “Managed” samples.</p>
</div>
<div id="heatmaps" class="section level2 hasAnchor">
<h2>7. Heatmaps<a href="module-3-microbiome-statistics-and-visualizations.html#heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Heatmaps visualize abundance patterns of taxa across samples or groups, allowing easy spotting of taxa that vary in abundance or co-occur across conditions. We can generate these heatmaps using <code>qiime feature-table heatmap</code>. However, raw feature tables at the ASV level often contain thousands of features, which can make heatmaps visually overwhelming and difficult to interpret. To create a clearer and more interpretable heatmap, you can collapse your feature table to a higher taxonomic rank, such as genus or family, before visualization, using the command below. <code>--p-level</code> sets the taxonomic rank, here we will collapse at the Class level.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb19-1" tabindex="-1"></a></span>
<span id="cb19-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb19-2" tabindex="-1"></a><span class="ex">qiime</span> taxa collapse <span class="dt">\</span></span>
<span id="cb19-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb19-3" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb19-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb19-4" tabindex="-1"></a>  <span class="at">--i-taxonomy</span> taxa/classification.qza <span class="dt">\</span></span>
<span id="cb19-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb19-5" tabindex="-1"></a>  <span class="at">--p-level</span> 3 <span class="dt">\</span></span>
<span id="cb19-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb19-6" tabindex="-1"></a>  <span class="at">--o-collapsed-table</span> deblur_output/table_class.qza</span></code></pre></div>
<p>Now we generate the heatmap using the collapsed class-level table.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb20-1" tabindex="-1"></a><span class="ex">qiime</span> feature-table heatmap <span class="dt">\</span></span>
<span id="cb20-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb20-2" tabindex="-1"></a>  <span class="at">--i-table</span> deblur_output/table_class.qza <span class="dt">\</span></span>
<span id="cb20-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb20-3" tabindex="-1"></a>  <span class="at">--m-sample-metadata-file</span> Blueberry_16S_metadata.tsv <span class="dt">\</span></span>
<span id="cb20-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb20-4" tabindex="-1"></a>  <span class="at">--m-sample-metadata-column</span> category <span class="dt">\</span></span>
<span id="cb20-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb20-5" tabindex="-1"></a>  <span class="at">--p-metric</span> braycurtis <span class="dt">\</span></span>
<span id="cb20-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb20-6" tabindex="-1"></a>  <span class="at">--p-color-scheme</span> RdYlBu <span class="dt">\</span></span>
<span id="cb20-7"><a href="module-3-microbiome-statistics-and-visualizations.html#cb20-7" tabindex="-1"></a>  <span class="at">--o-visualization</span> heatmap_class.qzv</span></code></pre></div>
<p>Open the resulting <code>heatmap_class.qzv</code> in QIIME 2 View to interactively explore taxa abundance patterns across samples.</p>
</div>
<div id="exporting-data" class="section level2 hasAnchor">
<h2>8. Exporting data from QIIME 2 for use in other software<a href="module-3-microbiome-statistics-and-visualizations.html#exporting-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>While QIIME 2 offers a wide range of tools for microbial community analysis, you may want to perform additional custom analyses in software like R, Python, or MATLAB. To do so, you’ll need to export your QIIME 2 artifacts into formats that are compatible with these platforms.</p>
<p>Representative sequences (i.e., ASVs) are stored in a <code>.qza</code> artifact that contains the DNA sequences used in downstream analysis. To export them into a standard FASTA format, use the following command:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb21-1" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="dt">\</span></span>
<span id="cb21-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb21-2" tabindex="-1"></a>   <span class="at">--input-path</span> deblur_output/rep_seqs_final.qza <span class="dt">\</span></span>
<span id="cb21-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb21-3" tabindex="-1"></a>   <span class="at">--output-path</span> deblur_output_exported</span></code></pre></div>
<p>Your sequences will be saved as <code>dna-sequences.fasta</code> inside the <code>deblur_output_16S_exported</code> folder. This file can be read by any downstream tool that accepts FASTA files.</p>
<p><a href="https://academic.oup.com/gigascience/article/1/1/2047-217X-1-7/2656152">BIOM</a> (Biological Observation Matrix) is a standardized format for representing feature tables, typically containing:
-Rows = features (e.g., ASVs, OTUs, taxa)
-Columns = samples
-Cells = abundance values (counts, relative abundances, etc.)
-Optional metadata (taxonomy, sample info)</p>
<p>BIOM files are widely used in microbiome analysis and supported by R packages like phyloseq, microbiome, and tools in Python such as scikit-bio and biom-format.To export a BIOM table (with taxonomy added as metadata) you can use the commands below.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-1" tabindex="-1"></a><span class="co">#First we fix taxonomy header with sed (required for biom add-metadata)</span></span>
<span id="cb22-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-2" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="at">-e</span> <span class="st">&#39;1 s/Feature/#Feature/&#39;</span> <span class="at">-e</span> <span class="st">&#39;1 s/Taxon/taxonomy/&#39;</span> taxa/taxonomy.tsv</span>
<span id="cb22-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-3" tabindex="-1"></a></span>
<span id="cb22-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-4" tabindex="-1"></a><span class="co">#Second we export the raw feature table and create the biom table</span></span>
<span id="cb22-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-5" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="dt">\</span></span>
<span id="cb22-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-6" tabindex="-1"></a>   <span class="at">--input-path</span> deblur_output/deblur_table_final.qza <span class="dt">\</span></span>
<span id="cb22-7"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-7" tabindex="-1"></a>   <span class="at">--output-path</span> deblur_output_exported</span>
<span id="cb22-8"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-8" tabindex="-1"></a></span>
<span id="cb22-9"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-9" tabindex="-1"></a><span class="co">#Third we add taxonomy metadata to the BIOM file</span></span>
<span id="cb22-10"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-10" tabindex="-1"></a><span class="ex">biom</span> add-metadata <span class="dt">\</span></span>
<span id="cb22-11"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-11" tabindex="-1"></a>   <span class="at">-i</span> deblur_output_exported/feature-table.biom <span class="dt">\</span></span>
<span id="cb22-12"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-12" tabindex="-1"></a>   <span class="at">-o</span> deblur_output_exported/feature-table_w_tax.biom <span class="dt">\</span></span>
<span id="cb22-13"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-13" tabindex="-1"></a>   <span class="at">--observation-metadata-fp</span> taxa/taxonomy.tsv <span class="dt">\</span></span>
<span id="cb22-14"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-14" tabindex="-1"></a>   <span class="at">--sc-separated</span> taxonomy</span>
<span id="cb22-15"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-15" tabindex="-1"></a><span class="co">#Last we convert the BIOM file to TSV format (tab-separated values)</span></span>
<span id="cb22-16"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-16" tabindex="-1"></a><span class="ex">biom</span> convert <span class="dt">\</span></span>
<span id="cb22-17"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-17" tabindex="-1"></a>   <span class="at">-i</span> deblur_output_exported/feature-table_w_tax.biom <span class="dt">\</span></span>
<span id="cb22-18"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-18" tabindex="-1"></a>   <span class="at">-o</span> deblur_output_exported/feature-table_w_tax.txt <span class="dt">\</span></span>
<span id="cb22-19"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-19" tabindex="-1"></a>   <span class="at">--to-tsv</span> <span class="dt">\</span></span>
<span id="cb22-20"><a href="module-3-microbiome-statistics-and-visualizations.html#cb22-20" tabindex="-1"></a>   <span class="at">--header-key</span> taxonomy</span></code></pre></div>
<p>This will give you a plain-text feature table (feature-table_w_tax.txt) with taxonomy annotations in the header row, which is especially useful for tools like R (phyloseq), Excel, or even manual inspection.</p>
<p>To export the tree of your ASVs in a <code>.nwk</code> format, use the command below.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb23-1" tabindex="-1"></a><span class="ex">qiime</span> tools export <span class="dt">\</span></span>
<span id="cb23-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb23-2" tabindex="-1"></a>  <span class="at">--input-path</span> asvs-tree.qza <span class="dt">\</span></span>
<span id="cb23-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb23-3" tabindex="-1"></a>  <span class="at">--output-path</span> deblur_output_exported</span></code></pre></div>
</div>
<div id="id_18s-its" class="section level2 hasAnchor">
<h2>9. Considerations for 18S and ITS data<a href="module-3-microbiome-statistics-and-visualizations.html#id_18s-its" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Most of the analyses described in this tutorial apply equally to 18S and ITS datasets, <strong>with one key exception</strong>: building a phylogenetic tree. Unlike 16S data, 18S and ITS amplicons generally lack a universally accepted reference phylogeny. Therefore, we must generate <em>de novo</em> phylogenetic trees.</p>
<p>To start, create symbolic links to the relevant data folders for your 18S and ITS datasets:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-1" tabindex="-1"></a><span class="bu">cd</span> ~/workspace/bmb_module3/</span>
<span id="cb24-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-2" tabindex="-1"></a><span class="fu">mkdir</span> 18S_analysis</span>
<span id="cb24-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-3" tabindex="-1"></a><span class="bu">cd</span> 18S_analysis</span>
<span id="cb24-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-4" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/18S_analysis/deblur_output .</span>
<span id="cb24-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-5" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/18S_analysis/taxa .</span>
<span id="cb24-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-6" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/18S_analysis/Plastisphere_18S_metadata.tsv .</span>
<span id="cb24-7"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-7" tabindex="-1"></a></span>
<span id="cb24-8"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-8" tabindex="-1"></a><span class="bu">cd</span> ~/workspace/bmb_module3/</span>
<span id="cb24-9"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-9" tabindex="-1"></a><span class="fu">mkdir</span> ITS_analysis</span>
<span id="cb24-10"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-10" tabindex="-1"></a><span class="bu">cd</span> ITS_analysis</span>
<span id="cb24-11"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-11" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/ITS_analysis/deblur_output .</span>
<span id="cb24-12"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-12" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/ITS_analysis/taxa .</span>
<span id="cb24-13"><a href="module-3-microbiome-statistics-and-visualizations.html#cb24-13" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module2/output/ITS_analysis/pregnancy_ITS_metadata.tsv .</span></code></pre></div>
<div id="building-a-de-novo-tree-for-18s" class="section level3 hasAnchor">
<h3>Building a de novo tree for 18S<a href="module-3-microbiome-statistics-and-visualizations.html#building-a-de-novo-tree-for-18s" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we’ll need to make a <em>de novo</em> multiple-sequence alignment of the ASVs using <a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb25-1" tabindex="-1"></a><span class="bu">cd</span> ~/workspace/bmb_module3/18S_analysis</span>
<span id="cb25-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb25-2" tabindex="-1"></a><span class="fu">mkdir</span> tree_out</span>
<span id="cb25-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb25-3" tabindex="-1"></a></span>
<span id="cb25-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb25-4" tabindex="-1"></a><span class="ex">qiime</span> alignment mafft <span class="at">--i-sequences</span> deblur_output/rep_seqs_final.qza <span class="dt">\</span></span>
<span id="cb25-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb25-5" tabindex="-1"></a>                      <span class="at">--p-n-threads</span> 4 <span class="dt">\</span></span>
<span id="cb25-6"><a href="module-3-microbiome-statistics-and-visualizations.html#cb25-6" tabindex="-1"></a>                      <span class="at">--o-alignment</span> tree_out/rep_seqs_final_aligned.qza</span></code></pre></div>
<p>Variable positions in the alignment need to be masked before FastTree is run, which can be done with the command below:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb26-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb26-1" tabindex="-1"></a><span class="ex">qiime</span> alignment mask <span class="at">--i-alignment</span>  tree_out/rep_seqs_final_aligned.qza <span class="dt">\</span></span>
<span id="cb26-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb26-2" tabindex="-1"></a>                     <span class="at">--o-masked-alignment</span>  tree_out/rep_seqs_final_aligned_masked.qza</span></code></pre></div>
<p>We finally run FastTree on this masked multiple-sequence alignment to make our tree with the command below:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb27-1" tabindex="-1"></a><span class="ex">qiime</span> phylogeny fasttree <span class="at">--i-alignment</span> tree_out/rep_seqs_final_aligned_masked.qza <span class="dt">\</span></span>
<span id="cb27-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb27-2" tabindex="-1"></a>                         <span class="at">--p-n-threads</span> 4 <span class="dt">\</span></span>
<span id="cb27-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb27-3" tabindex="-1"></a>                         <span class="at">--o-tree</span> tree_out/rep_seqs_final_aligned_masked_tree</span></code></pre></div>
<p>FastTree returns an unrooted tree. One basic way to add a root to a tree is to add it at the midpoint of the largest tip-to-tip distance in the tree, which is done with this command:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb28-1" tabindex="-1"></a><span class="ex">qiime</span> phylogeny midpoint-root <span class="at">--i-tree</span> tree_out/rep_seqs_final_aligned_masked_tree.qza <span class="dt">\</span></span>
<span id="cb28-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb28-2" tabindex="-1"></a>                              <span class="at">--o-rooted-tree</span> tree_out/rep_seqs_final_aligned_masked_tree_rooted.qza</span></code></pre></div>
<p>Let’s rename the tree and place it in our analysis folder.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb29-1" tabindex="-1"></a><span class="fu">cp</span> tree_out/rep_seqs_final_aligned_masked_tree_rooted.qza asvs-tree.qza</span></code></pre></div>
<p><strong>Due to memory or time constraints, you may opt to use a tree that has already been computed. Use the following command to create symbolic links to precomputed trees</strong></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="module-3-microbiome-statistics-and-visualizations.html#cb30-1" tabindex="-1"></a><span class="bu">cd</span> ~/workspace/bmb_module3/18S_analysis</span>
<span id="cb30-2"><a href="module-3-microbiome-statistics-and-visualizations.html#cb30-2" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module3/output/18S_analysis/asvs-tree.qza .</span>
<span id="cb30-3"><a href="module-3-microbiome-statistics-and-visualizations.html#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="module-3-microbiome-statistics-and-visualizations.html#cb30-4" tabindex="-1"></a><span class="bu">cd</span> ~/workspace/bmb_module3/ITS_analysis</span>
<span id="cb30-5"><a href="module-3-microbiome-statistics-and-visualizations.html#cb30-5" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> ~/CourseData/MIC_data/bmb_module3/output/ITS_analysis/asvs-tree.qza .</span></code></pre></div>
</div>
<div id="building-the-its-tree" class="section level3 hasAnchor">
<h3>Building the ITS tree<a href="module-3-microbiome-statistics-and-visualizations.html#building-the-its-tree" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Repeat the same commands shown above, but make sure you use the <code>deblur_output</code> from the ITS analysis. This will generate a <em>de novo</em> tree for your ITS dataset.</p>
<p>Once you have generated or linked the <em>de novo</em> trees for your 18S and ITS data, you can proceed with all downstream analyses — including diversity metrics, ordinations, and statistical testing — just as you would with 16S data.</p>
</div>
</div>
<div id="answers" class="section level2 hasAnchor">
<h2>Answers<a href="module-3-microbiome-statistics-and-visualizations.html#answers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Question 1: What is a good rarefaction depth for diversity analysis?</strong></p>
<p>A cut-off of 4,000 reads will be sufficient: the curve plateaus around this depth and we won’t exclude any samples.</p>
<p><strong>Question 2: are there any significant differences in alpha diversity between any of our metadata categories?</strong></p>
<p>There are significant differences in richness and phylogenetic diversity between forest and managed environment samples. There are no significant differences between bulk and rhizosphere soil.</p>
<p><strong>Question 3: which metadata category appears to provide more separation in the beta diversity PCoA plots?</strong></p>
<p>This is hard to say just by looking at the Emperor plots provided by QIIME2, but the forest/managed category appears to exhibit more distinct separation. The PERMANOVA test from the <code>beta-group-significance</code> command shows this as well.</p>
<p><strong>Question 4: what do you mean I’m getting an error message? It should be working fine, it worked last time! I can’t see what I’m doing wro… oooooh there we go, I see now :)</strong></p>
<p>There is a typo in <code>--m-metadata-column Category</code>: The “C” should be lowercase, matching the column header in your metadata file. ;)</p>
<p><strong>Question 5: can you identify any patterns between the metadata groups?</strong></p>
<p>Because stacked barcharts are limited in their analytical capabilities, it is hard to discern anything except very obvious patterns.</p>
<p><strong>Question 6: Does ANCOM identify any differentially abundant taxa between forest and managed environments?</strong></p>
<p>One ASVs is identified as differentially abundant between forest and managed environments using ANCOM. You can look up the identity of each ASV in the taxonomy.tsv file. With ANCOM-BC many more ASVs are identified as diferentially abundant.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-1.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/cbw-dev/bookdown-template/blob/main/040-module-3.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
